packages/core/src/html/scriptAndStyleTagProcessor.js

```diff
@@ -1,44 +1,23 @@
-/**
- * Adds attribute "type=application/javascript" to <script> tag.
- *
- * This is necessary because we are pre-compiling each page as a Vue application,
- * where the Vue compilation ignores and discards the <script> tag.
- *
- * By having this attribute, the compilation will ignore the <script> tag but not discard it.
- *
- * @param {Object<any, any>} node from the dom traversal
- */
-function processScriptTag(node) {
-  node.attribs.type = 'application/javascript';
-}
+const cheerio = require('cheerio');
 
 /**
- * Changes <style> tag into placeholder tag:
- * <script src defer type="application/javascript" style-bypass-vue-compilation>.
- *
- * This is necessary because we are pre-compiling each page as a Vue application,
- * where the Vue compilation ignores and discards the <style> tag.
+ * Removes every script and style node (written by the user) encountered in the main app,
+ * and records all these script nodes in an array so that they can be hoisted to after
+ * <body> tag at a later stage.
  *
- * We work around this problem by piggybacking on the approach we used for <script> tags to be ignored
- * and not discarded; by changing <style> tags to the placeholder tag as shown above.
- * We will change it back to <style> tag after Vue compilation and after the element is created by Vue.
+ * This is to prevent Vue compilation of script/style tags (as they are not meant to be compiled).
  *
  * @param {Object<any, any>} node from the dom traversal
  */
-function processStyleTag(node) {
-  node.name = 'script';
-  /*
-    This node is not intended to be ran as a script. Thus, we have to defer the execution of the script
-    until the node is restored as a style node (which is handled by MarkBind's script).
-    The defer attribute requires the src attribute to work. Thus, the src attribute is used as a dummy.
-  */
-  node.attribs.src = '';
-  node.attribs.defer = '';
-  node.attribs.type = 'application/javascript'; // to bypass Vue compilation as a script tag
-  node.attribs['style-bypass-vue-compilation'] = ''; // to allow specific query selection of this element
+function processScriptAndStyleTag(node, userScriptsAndStyles) {
+  const idx = node.parent.children.indexOf(node);
+  if (idx !== -1) {
+    node.parent.children.splice(idx, 1);
```

Comment:
just to double check -- does this cause issues with sibling traversal?

```
<script>...</script>
<panel>this actually gets processed</panel>
```

Reply:
Thanks for bringing up this point! As discussed offline, it should be fine as our tests are currently passing.